core/jazz_t-vault/index.js

```diff
@@ -0,0 +1,410 @@
+          return cb(null, responseObj(result, event.body));
+        })
+        .catch(err => {
+          logger.error("Error creating safe: " + JSON.stringify(err));
+          if (err.errorType && err.errorType === "inputError") return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+          else return cb(JSON.stringify(errorHandler.throwInternalServerError(err.error)));
+        });
+    }
+
+    else if (event && event.method === 'GET' && resourcePath.endsWith("/safes/{safename}")) {
+      validations.validateSafeInput(event)
+        .then(() => { return validations.genericInputValidation(event.path) })
+        .then(() => { return vault.getVaultToken(configData) })
+        .then((vaultToken) => { return exportable.getSafeDetails(event.path.safename, configData, vaultToken) })
+        .then(result => {
+          logger.info("Successfully got safe details");
+          return cb(null, responseObj(result, event.path));
+        })
+        .catch(err => {
+          logger.error("Error getting safe details: " + JSON.stringify(err));
+          if (err.errorType && err.errorType === "inputError") return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+          else return cb(JSON.stringify(errorHandler.throwInternalServerError(err.error)));
+        });
+    }
+
+    else if (event && event.method === 'PUT' && resourcePath.endsWith("/safes/{safename}")) {
+      validations.validateSafeInput(event)
+        .then(() => { return validations.validateUpdateSafeInput(event) })
+        .then(() => { return validations.genericInputValidation(event.body) })
+        .then(() => { return validations.genericInputValidation(event.path) })
+        .then(() => { return vault.getVaultToken(configData) })
+        .then((vaultToken) => { return exportable.updateSafe(event.body, event.path.safename, configData, vaultToken) })
+        .then(result => {
+          logger.info("Successfully updated safe");
+          return cb(null, responseObj(result, event.body));
+        })
+        .catch(err => {
+          logger.error("Error updating safe: " + JSON.stringify(err));
+          if (err.errorType && err.errorType === "inputError") return cb(JSON.stringify(errorHandler.throwInputValidationError(err.message)));
+          else return cb(JSON.stringify(errorHandler.throwInternalServerError(err.error)));
+        });
+    }
+
+    else if (event && event.method === 'DELETE' && resourcePath.endsWith("/safes/{safename}")) {
+      validations.validateSafeInput(event)
+        .then(() => { return validations.genericInputValidation(event.path) })
+        .then(() => { return vault.getVaultToken(configData) })
+        .then((vaultToken) => { return exportable.deleteSafe(event.path.safename, configData, vaultToken) })
+        .then(result => {
+          logger.info("Successfully deleted safe");
+          return cb(null, responseObj(result, event.path));
```

Comment:
Why are we returning `event.path` here?

Reply:
We are returning the responses in the format of { data: {}, input:{}} , since here  we are taking safename  in the path as input, returning that in the response object.